### PR TITLE
configs/imxrt1050-evk/loadable_elf_apps : Generate stripped ELF

### PR DESCRIPTION
**Pull request by kishore-sn**
Add -s flag to linker flags to generated stripped ELF binary

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [ea833a4240c031fb7ec497ec63932c767945cfbd] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ea833a4240c031fb7ec497ec63932c767945cfbd] - Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@kishore-sn Could you apply this like below?
```
ifeq ($(CONFIG_DEBUG_SYMBOLS),y)
  LDFLAGS += -g
endif
```

**Comment by sunghan-chang** [comment]
And could you add that at all of Make.defs under configs?

File: build/configs/imxrt1050-evk/loadable_elf_apps/Make.defs [hunk 1]
```diff
@@ -139,7 +139,7 @@ AFLAGS = $(CFLAGS) -D__ASSEMBLY__
 CELFFLAGS = $(CFLAGS) -mlong-calls # --target1-abs
 CXXELFFLAGS = $(CXXFLAGS) -mlong-calls # --target1-abs
 
-LDELFFLAGS = -r -e main
+LDELFFLAGS = -r -e main -s
```
Comment (btheosam) [inline build/configs/imxrt1050-evk/loadable_elf_apps/Make.defs]: Better to put inside a flag and if dynamic linking is enabled, then this flag should n't be there. Account  these.

**Comment by kishore-sn** [comment]
I made the below changes:
- Check for CONFIG_DEBUG_SYMBOLS and add -s option
- Disable CONFIG_DEBUG_SYMBOLS in defconfig

I have made these changes only in loadable_elf_apps config because LDELFFLAGS is defined only in this config.

**Comment by seinfra** [comment]
Target : [ea303a9683de21fb4809d651b117faaae730de00] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ea303a9683de21fb4809d651b117faaae730de00] - Code Rule Check (C++) OK.

**Comment by sunghan-chang** [comment]
@kishore-sn What I wrote is an example. Is it appropriatary with CONFIG_DEBUG_SYMBOLS?

**Comment by kishore-sn** [comment]
Ok. I understand. 
I added a new config for this.

**Comment by seinfra** [comment]
Target : [9e8ade1a9aa0e646fc85bc41641e3dd6e505d42f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9e8ade1a9aa0e646fc85bc41641e3dd6e505d42f] - Code Rule Check (C++) OK.

File: os/binfmt/libelf/Kconfig [hunk 2]
```diff
@@ -48,3 +48,10 @@ config ELF_EXIDX_SECTNAME
 		be loaded by the ELF binary loader.
 
 		This is needed to support exception handling on loadable ELF modules.
+
+config ELF_INCLUDE_SYMBOLS
```
Comment (sunghan-chang) [inline os/binfmt/libelf/Kconfig]: 1. How about `exclude` and `ifeq ($(CONFIG_ELF_EXCLUDE_SYMBOLS),y)`?
`include` looks positive but real operation `ifneq` is negative.
I think exclude is better to understand.
2.  Is this proper location to add? Here is not a location of making binary. Isn't it?
3. Symbol information is necessary to debug using T32. Then, how about making a new binary which is stripped?
I mean after making non-stripped elf and then making new stripped file to download.

File: build/configs/imxrt1050-evk/loadable_elf_apps/Make.defs [hunk 3]
```diff
@@ -140,6 +140,9 @@ CELFFLAGS = $(CFLAGS) -mlong-calls # --target1-abs
 CXXELFFLAGS = $(CXXFLAGS) -mlong-calls # --target1-abs
 
 LDELFFLAGS = -r -e main
+ifneq ($(CONFIG_ELF_INCLUDE_SYMBOLS),y)
```
Comment (sunghan-chang) [inline build/configs/imxrt1050-evk/loadable_elf_apps/Make.defs]: Could you add this at every Make.defs?

File: build/configs/imxrt1050-evk/loadable_elf_apps/defconfig [hunk 4]
```diff
@@ -823,6 +822,7 @@ CONFIG_ELF_ALIGN_LOG2=2
 CONFIG_ELF_STACKSIZE=2048
 CONFIG_ELF_BUFFERSIZE=32
 CONFIG_ELF_BUFFERINCR=32
+# CONFIG_ELF_INCLUDE_SYMBOLS is not set
```
Comment (sunghan-chang) [inline build/configs/imxrt1050-evk/loadable_elf_apps/defconfig]: You must keep a file permission with 644. Please revert

File: os/binfmt/libelf/Kconfig (hunk 2 above)
Comment (sunghan-chang) [reply os/binfmt/libelf/Kconfig]: for 3rd, Cypress does like what I said.

File: build/configs/imxrt1050-evk/loadable_elf_apps/Make.defs (hunk 3 above)
Comment (sunghan-chang) [reply build/configs/imxrt1050-evk/loadable_elf_apps/Make.defs]: I saw your comment now. Please ignore this.

**Comment by kishore-sn** [comment]
@sunghan-chang  I made below changes
1. Changed the config to CONFIG_ELF_EXCLUDE_SYMBOLS
2. Moved the logic from Make.defs to Makefile
3. Added logic to keep both stripped and non-stripped binary files
4. The non stripped file will be named as micom_dbg  and the stripped binary is named as micom
5. Changed file permission

**Comment by seinfra** [comment]
Target : [d9f4a24539193fac81f1d6d8f798ecae74ee2fc1] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d9f4a24539193fac81f1d6d8f798ecae74ee2fc1] - Code Rule Check (C++) OK.

File: build/configs/imxrt1050-evk/loadable_elf_apps/defconfig [hunk 5]
```diff
@@ -285,7 +285,6 @@ CONFIG_ARCH_HAVE_BUTTONS=y
 # CONFIG_ARCH_BUTTONS is not set
```
Comment (sijupunnnoose) [inline build/configs/imxrt1050-evk/loadable_elf_apps/defconfig]: 100755->100644

File: apps/examples/elf/micomapp/Makefile [hunk 6]
```diff
@@ -113,6 +113,10 @@ clean:
 install:
 	$(Q) mkdir -p $(USER_BIN_DIR)
 	$(Q) install $(BIN) $(USER_BIN_DIR)/$(BIN)
+ifeq ($(CONFIG_ELF_EXCLUDE_SYMBOLS),y)
```
Comment (sunghan-chang) [inline apps/examples/elf/micomapp/Makefile]: I suggest two location to add this.
1. Makefile.unix
```
...
ifeq ($(CONFIG_RAW_BINARY),y)
	$(Q) echo "CP: $(BIN_EXE).bin"
	$(Q) $(OBJCOPY) $(OBJCOPYARGS) -O binary $(BIN) $(BIN).bin
endif
HERE!!!!!!!!!!!!!!!!!!!!!!!!!
	$(Q) echo "Start the Board Specific Work for Binary"
	$(Q) $(call MAKE_BOARD_SPECIFIC_BIN, $(BIN_EXE), $(BIN_EXT))
...
```
2. In MAKE_BOARD_SPECIFIC_BIN definition of each board.

**Comment by seinfra** [comment]
Target : [2d3b38d3ad5ad1cf618f0e4add32d9cc82f1d40a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2d3b38d3ad5ad1cf618f0e4add32d9cc82f1d40a] - Code Rule Check OK.

File: apps/examples/elf/micomapp/Makefile (hunk 6 above)
Comment (kishore-sn) [reply apps/examples/elf/micomapp/Makefile]: We have below line in the micomapp/Makefile and wifiapp/Makefile

`$(Q) $(TOPDIR)/tools/mkbinheader.py $(USER_BIN_DIR)/$(BIN) $(BIN_TYPE) $(KERNEL_VER) $(BIN) $(BIN_VER) $(DYNAMIC_RAM_SIZE)`

It is not possible to apply the strip command after adding the header. So, I need to make these changes in these Makefile's only.

**Pull request merged**